### PR TITLE
Convert @stripe/stripe-js to a peer dependency

### LINKED ISSUE #110: types are incompatible with latest @stripe/stripe-js@3.3.0
### Describe the bug

Now svelte-stripe uses `@stripe/stripe-js@2.4.0` under the hood. But if I pass `stripe` object of type `Stripe` imported from `@stripe/stripe-js@3.3.0`, used in my project, I receive the following (poorly readable) ts error: 

```
Type 'import("/node_modules/@stripe/stripe-js/dist/stripe-js/stripe").Stripe' is not assignable to type 'import("/node_modules/svelte-stripe/node_modules/@stripe/stripe-js/types/stripe-js/stripe").Stripe'.
  The types of 'elements(...).create' are incompatible between these types.
    Type '{ (elementType: "address", options: import("/node_modules/@stripe/stripe-js/dist/stripe-js/elements/address").StripeAddressElementOptions): import("/node_modules/@stripe/stripe-js/dist/stripe-js/elements/address").StripeAddressElement; (elementType: "paym...' is not assignable to type '{ (elementType: "address", options: import("/node_modules/svelte-stripe/node_modules/@stripe/stripe-js/types/stripe-js/elements/address").StripeAddressElementOptions): import("/node_modules/svelte-stripe/node_modules/@stripe/stripe-js/types/stripe-js/elem...'.
      Types of parameters 'elementType' and 'elementType' are incompatible.
        Type '"cart"' is not assignable to type '"address"'.
```

```svelte
<script lang="ts">
  import { loadStripe, type Stripe } from '@stripe/stripe-js'
  import { Elements } from 'svelte-stripe'
  import { onMount } from 'svelte'

  let stripe: Stripe | null

  onMount(async () => {
    stripe = await loadStripe('abc')
  })
</script>

{#if stripe}
  <Elements {stripe}>
    <!--  Error here -->
  </Elements>
{/if}
```

It is difficult to workaround this issue (svelte does not allow `ts-ignore` inside markup). Also ignoring ts errors is not a best workaround :)

### Reproduction

I made a reproduction repo with this error. Make `npm install` and look into `src/+page.svelte` file OR run `npm run check`
https://github.com/stanf0rd/svelte-stripe-ts-error-repro

### Severity

annoyance

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by hmnd**
fixes #110 

- update @stripe/stripe-js to v3 and changes to a dev+peer dep
- update docs to reflect @stripe/stripe-js requirement

### CODE REVIEW

**Comment by joshnuss** [comment]
Thanks @hmnd!

Are eslint updates and dependency updates related to updating the peer dependency? Or can they be in a separate PR?

**Comment by hmnd** [comment]
Can split them out into a separate PR!

File: src/routes/+page.md [hunk 1]
```diff
@@ -17,13 +17,14 @@ Links:
...
+To configure your project, add these 3 packages:
 
 ```bash
-pnpm install -D stripe svelte-stripe
+pnpm install -D stripe @stripe/stripe-js svelte-stripe
 ```
 
-- [stripe](https://npmjs.org/package/stripe) is the official server-side version of Stripe.
+- [stripe](https://npmjs.org/package/stripe) is Stripe's official server-side library.
+- [@stripe/stripe-js](https://www.npmjs.com/package/@stripe/stripe-js) is Stripe's official client-side library, which always loads the latest version of [Stripe.js](https://stripe.com/docs/stripe-js) to ensure [PCI compliance](https://stripe.com/docs/security/guide#validating-pci-compliance).
```
Comment (joshnuss) [inline src/routes/+page.md:27]: I would shorten it to "is Stripe's official client-side library."

My reasoning is to keep the docs as short as possible, so people don't have to read too much to get running.

File: src/routes/+page.md [hunk 2]
```diff
@@ -105,7 +106,7 @@ To use it, drop a `<PaymentElement>` component in your form:
 ```html
 <form on:submit|preventDefault="{submit}">
   <Elements {stripe} {clientSecret} bind:elements>
-    <PaymentElement options={...} />
+    <PaymentElement options="{...}" />
```
Comment (joshnuss) [inline src/routes/+page.md:109]: If possible, let's keep linting changes as a separate PR.

**Review by joshnuss — COMMENTED** [review]
Looks great! Left 2 small comments

File: src/routes/+page.md (hunk 2 above)
Comment (hmnd) [reply src/routes/+page.md:109]: Sorry, must've autoformatted on save

Comment (joshnuss) [reply src/routes/+page.md:109]: No worries! Happens to me all the time too 😅

File: vite.config.js [hunk 3]
```diff
@@ -3,5 +3,9 @@ import basicSsl from '@vitejs/plugin-basic-ssl'
 import { defineConfig } from 'vite'
 
 export default defineConfig({
-  plugins: [sveltekit(), basicSsl()]
+  plugins: [sveltekit(), basicSsl()],
+  server: {
+    // temp fix for https://github.com/sveltejs/kit/issues/11365#issuecomment-1979797510
+    proxy: {}
+  }
```
Comment (joshnuss) [inline vite.config.js:10]: I'm not I understand why this needed? 
Is it not working in dev mode with latest Node?

Comment (hmnd) [reply vite.config.js:10]: It wasn't for me; I was getting the error in the linked thread. But if you'd prefer to leave that out, I can get rid of it :)

Comment (joshnuss) [reply vite.config.js:10]: Yeah, let's leave it out. Cause I'd need to review that issue

**Review by joshnuss — APPROVED** [review]

**Comment by joshnuss** [comment]
Thanks @hmnd!! Much appreciated
